### PR TITLE
Added swiftformat configuration

### PR DESCRIPTION
**Pull request by ckaik**
I've created basic swiftformat + swiftlint configs. Got both of them form [here](https://github.com/gurhub/dotfiles/blob/master) but made some modifications.

Both can be used from the `Makefile`, instructions are in the README.

It didn't really change all that much. Mostly there's more wrapping for initializers, cause some of them are _really_ long. Also it does remove the File Header, that can be disabled though, I just enabled that option because I personally prefer my files without the headers.

One thing that can't be fixed using these tools would be the naming convention of parameters. Snake case is being used a lot right now but camel case would make more sense for Swift. That is a big undertaking though.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +53,21 @@ Luckily this just scratches the surface of what you can do with the Lemmy Swift
 
 Lemmy Swift Client is available under the MIT license. See the [LICENSE](https://github.com/rrainn/Lemmy-Swift-Client/blob/main/LICENSE).
 
+## Code Formatting + Linting
+
+This project uses [swiftformat](https://github.com/nicklockwood/SwiftFormat) for code formatting as well as [swiftlint](https://github.com/realm/SwiftLint) for linting. 
+Please make sure to format and lint your code before submitting a Pull Request.
```
Comment (fishcharlie) [inline README.md]:
```suggestion
This project uses [swiftformat](https://github.com/nicklockwood/SwiftFormat) for code formatting as well as [swiftlint](https://github.com/realm/SwiftLint) for linting.

Please make sure to format and lint your code before submitting a Pull Request.
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+lint:
+	swiftlint --fix
```
Comment (fishcharlie) [inline Makefile]: What if we were to have a `lint` and a `lint-fix` or something like that?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+format:
+	swiftformat .
```
Comment (fishcharlie) [inline Makefile]: Does this automatically fix invalid formatting? Or does it just throw an error if invalid formatting is detected?

**Review by fishcharlie — CHANGES_REQUESTED** [review]
Overall this looks good. Couple of code level comments.

One more high level thing. I think it'd be useful to have a GitHub action that runs both the lint and format, but does not actually fix anything.

That way it ensures that any PR made actually conforms to the style before merging.

File: Makefile (hunk 2 above)
Comment (ckaik) [reply Makefile]: Thought about that, shouldn't be an issue.

File: Makefile (hunk 3 above)
Comment (ckaik) [reply Makefile]: This one automatically reformats. There's a `--dryrun` option one could use to just check for issues.

**Comment by ckaik** [comment]
I've added a new `lint-fix` target. The existing `lint` target now just lists the issues.

**Comment by fishcharlie** [comment]
@ckaik If we do a `--dryrun` option for `make format` and rename the existing one to `make format-fix`. Then we should be able to add a GitHub Action here. Something like this maybe?

```yml
name: CI

on:
  push:
  pull_request:

jobs:
  lint:
    runs-on: macos-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2
      - name: Lint code
        run: make lint
      - name: Format code
        run: make format
```

**Comment by ckaik** [comment]
Honestly I haven't used Github actions yet so I'm not familiar. It looks fine though ;) I did add the dryrun option to the format target.

Not sure if the swiftformat output is _really_ useful though. It only shows the rules would have been applied and which files they would apply to, no specifics.

Lint is more useful.

**Comment by fishcharlie** [comment]
@ckaik I think it depends on the exit code. If the exit code is 0, it'll pass, and if it's anything else it'll fail.

So I guess we need to verify that `swiftformat` actually does return an exit code >0 on dryrun if changes need to be applied.

I can work on the GitHub action piece later.

**Comment by ckaik** [comment]
Made another update. The `--dryun` argument doesn't return non-zero if it fails. However the `--lint` option _does_. So I added that instead.

Though now I'm not sure if both swiftformat _and_ swiftlint are needed

**Comment by fishcharlie** [comment]
@ckaik The thing I was most concerned about was enforcing consistency with tabs vs spaces. So if we need to remove one to simplify things, that would be fine by me, so long as that main objective is achieved.

**Comment by ckaik** [comment]
Then I would propose to remove `swiftlint`, also:

* change the `format` target to run swiftformat like before, fixing issues
* change the lint target to execute `swiftformat --lint .` - that'll return an error code if it doesn't pass

Tab indentation is a rule in swiftformat.

**Comment by ckaik** [comment]
Made the changes.

* Removed `.swiftlint.yml`
* `make format` now runs `swiftformat .`, formatting all code
* `make lint` now runs `swiftformat --lint .`, checking the formatting rules for all code
* Adjusted README

So your proposed github action would now only need to run `make lint`.

**Comment by ckaik** [comment]
One more update, I just reverted the actual reformatting of the whole project.

I believe this should be done after merging. 
This will make the PR easier to digest.

**Review by fishcharlie — APPROVED** [review]

**Pull request merged**